worker-taskstowing/src/main/java/com/microfocus/caf/worker/taskstowing/database/HikariDataSourceFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021 Micro Focus or one of its affiliates.
+ *
+ * The only warranties for products and services of Micro Focus and its
+ * affiliates and licensors ("Micro Focus") are set forth in the express
+ * warranty statements accompanying such products and services. Nothing
+ * herein should be construed as constituting an additional warranty.
+ * Micro Focus shall not be liable for technical or editorial errors or
+ * omissions contained herein. The information contained herein is subject
+ * to change without notice.
+ *
+ * Contains Confidential Information. Except as specifically indicated
+ * otherwise, a valid license is required for possession, use or copying.
+ * Consistent with FAR 12.211 and 12.212, Commercial Computer Software,
+ * Computer Software Documentation, and Technical Data for Commercial
+ * Items are licensed to the U.S. Government under vendor's standard
+ * commercial license.
+ */
+package com.microfocus.caf.worker.taskstowing.database;
+
+import com.microfocus.caf.worker.taskstowing.factory.TaskStowingWorkerConfiguration;
+import com.zaxxer.hikari.HikariDataSource;
+
+final class HikariDataSourceFactory
+{
+    public static HikariDataSource createHikariDataSource(final TaskStowingWorkerConfiguration configuration)
+    {
+        final HikariDataSource hikariDataSource = new HikariDataSource();
+
+        final String jdbcUrl = String.format(
+            "jdbc:postgresql://%s:%s/%s?ApplicationName=%s",
```

Comment:
I'm using the Hikari connection pool in order to set a max pool size on the connections.

I noticed on Simran's PR you mentioned using Properties to set the JDBC URL. I tried that here but doesn't seem to work with Hikari:

```
 final class HikariDataSourceFactory
 {
     public static HikariDataSource createHikariDataSource(final TaskStowingWorkerConfiguration configuration)
     {
         final Properties properties = new Properties();
         PGProperty.PG_HOST.set(properties, configuration.getDatabaseHost());
         PGProperty.PG_PORT.set(properties, configuration.getDatabasePort());
         PGProperty.PG_DBNAME.set(properties, configuration.getDatabaseName());
         PGProperty.APPLICATION_NAME.set(properties, configuration.getDatabaseAppName());
         PGProperty.USER.set(properties, configuration.getDatabaseUsername());
         PGProperty.PASSWORD.set(properties, configuration.getDatabasePassword());
 
         final HikariConfig hikariConfig = new HikariConfig(properties);
         hikariConfig.setMaximumPoolSize(configuration.getDatabaseMaximumPoolSize());        
 
         return new HikariDataSource(hikariConfig);
     }
 
     private HikariDataSourceFactory()
     {
     }
 }
```

Returns this error:
 
 ```
 [17:20:18.200Z #c9b.001 ERROR -            -   ] c.z.hikari.util.PropertyElf: Property PGHOST does not exist on target class com.zaxxer.hikari.HikariConfig
java.lang.RuntimeException: Property PGHOST does not exist on target class com.zaxxer.hikari.HikariConfig
	at com.zaxxer.hikari.util.PropertyElf.setProperty(PropertyElf.java:135)
	at com.zaxxer.hikari.util.PropertyElf.lambda$setTargetFromProperties$0(PropertyElf.java:59)
	at java.base/java.util.concurrent.ConcurrentHashMap.forEach(ConcurrentHashMap.java:1603)
	at java.base/java.util.Properties.forEach(Properties.java:1443)
	at com.zaxxer.hikari.util.PropertyElf.setTargetFromProperties(PropertyElf.java:54)
	at com.zaxxer.hikari.HikariConfig.<init>(HikariConfig.java:140)
	at com.microfocus.caf.worker.taskstowing.database.HikariDataSourceFactory.createHikariDataSource(HikariDataSourceFactory.java:39)
	at com.microfocus.caf.worker.taskstowing.database.DatabaseClient.<init>(DatabaseClient.java:38)
	at com.microfocus.caf.worker.taskstowing.factory.TaskStowingWorkerFactory.<init>(TaskStowingWorkerFactory.java:69)
	at com.microfocus.caf.worker.taskstowing.factory.TaskStowingWorkerFactoryProvider.getWorkerFactory(TaskStowingWorkerFactoryProvider.java:34)
	at com.hpe.caf.worker.core.WorkerApplication.run(WorkerApplication.java:108)
	at com.hpe.caf.worker.core.WorkerApplication.run(WorkerApplication.java:59)
	at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:44)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:87)
	at io.dropwizard.cli.Cli.run(Cli.java:78)
	at io.dropwizard.Application.run(Application.java:94)
	at com.hpe.caf.worker.core.WorkerApplication.main(WorkerApplication.java:74)
	```

Reply:
I haven't been able to get this to work, tried:

```
hikariDataSource.setDataSourceClassName("org.postgresql.ds.PGSimpleDataSource");
hikariDataSource.addDataSourceProperty(PGProperty.PG_HOST.getName(), configuration.getDatabaseHost());
hikariDataSource.addDataSourceProperty(PGProperty.PG_PORT.getName(), configuration.getDatabasePort());
hikariDataSource.addDataSourceProperty(PGProperty.PG_DBNAME.getName(), configuration.getDatabaseName());
hikariDataSource.addDataSourceProperty(PGProperty.APPLICATION_NAME.getName(), configuration.getDatabaseAppName());
hikariDataSource.addDataSourceProperty(PGProperty.USER.getName(), configuration.getDatabaseUsername());
hikariDataSource.addDataSourceProperty(PGProperty.PASSWORD.getName(), configuration.getDatabasePassword());
```

but this is throwing the error:

```
Property PGHOST does not exist on target class org.postgresql.ds.PGSimpleDataSource
```

Reply:
I believe these are Driver properties.  Try using `hikariDataSource.setDriverClassName("org.postgresql.Driver");` instead of `hikariDataSource.setDataSourceClassName("org.postgresql.ds.PGSimpleDataSource");`.

Reply:
No luck either:

```
java.lang.IllegalArgumentException: jdbcUrl is required with driverClassName.\n\tat com.zaxxer.hikari.HikariConfig.validate(HikariConfig.java:1025
```

```
hikariDataSource.setDriverClassName("org.postgresql.Driver");
hikariDataSource.addDataSourceProperty(PGProperty.PG_HOST.getName(), configuration.getDatabaseHost());
hikariDataSource.addDataSourceProperty(PGProperty.PG_PORT.getName(), configuration.getDatabasePort());
hikariDataSource.addDataSourceProperty(PGProperty.PG_DBNAME.getName(), configuration.getDatabaseName());
hikariDataSource.addDataSourceProperty(PGProperty.APPLICATION_NAME.getName(), configuration.getDatabaseAppName());
hikariDataSource.addDataSourceProperty(PGProperty.USER.getName(), configuration.getDatabaseUsername());
hikariDataSource.addDataSourceProperty(PGProperty.PASSWORD.getName(), configuration.getDatabasePassword());
hikariDataSource.setMaximumPoolSize(configuration.getDatabaseMaximumPoolSize());
```

Reply:
So include a JDBC Url as well, but just make it `"jdbc:postgresql:"`.  In fact you may not need to set driver class name if you do that because it can probably derive it from the `postgresql` part of the url.

Note that the issue being resolved here is avoiding the building up of a url - there's no issue using a constant one and supplying the variables separately.

Reply:
Thanks, that's it now.